### PR TITLE
Build docker image and deploy to Kubernetes

### PR DESCRIPTION
**Pull request by CMDRyan**
@Steven-Chan 

- Escalated console.log to console.warn
- Set default port number
- Add logging function if mongoDB is available 

- Added Dockerfile to build image
- Added Docker-compose for both app.js and mongoDB

- Created deployment.yaml, service.yaml, ingress.yaml resources for Kubernetes
- Created helm-chart for all resources.

- Removed app id and secret from files.
- Updated README instruction

### CODE REVIEW

File: helm-chart/values.yaml [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+  # requests:
+  #   cpu: 100m
+  #   memory: 128Mi
+
+autoscaling:
+  enabled: false
+  minReplicas: 1
+  maxReplicas: 100
+  targetCPUUtilizationPercentage: 80
+  # targetMemoryUtilizationPercentage: 80
```
Comment (Steven-Chan) [inline helm-chart/values.yaml:48]: Seems the template is not using it, pls remove this section.

File: Dockerfile [hunk 2]
```diff
@@ -0,0 +1,19 @@
+FROM node:12.20.0
+
+WORKDIR /usr/src/app
+COPY package*.json ./ 
+
+ENV COOKIE_AGE=86400000
+
+RUN yarn
+
+COPY yarn.lock ./
```
Comment (Steven-Chan) [inline Dockerfile:10]: This is in wrong order and a bit confusing.

1. We use yarn, so we just need `package.json` and `yarn.lock`.
2. The lock file is required to install determined version of dependencies.

**Review by Steven-Chan — CHANGES_REQUESTED** [review]
Please create a Makefile that contains the following command.

```
make build -> build the docker image
make push -> push the docker image to a predefined container registry
```

**Comment by CMDRyan** [comment]
@Steven-Chan 

- Removed autoscaling lines in values.yaml
- Reorganized Dockerfile Build order
- Created makefileconfig.env to define app name and registry
- Created makefile to build and push docker image using values from makefileconfig.env

**Pull request merged**